sashimi-webapp/src/logic/documentPackager/conditionalProcessor.js

```diff
@@ -0,0 +1,240 @@
+
+  if (silent) { return false; } // don't run any pairs in validation mode
+  if (start + 8 >= max) { return false; }
+
+  if (state.src.charCodeAt(start) !== 61 /* = */ ||
+      state.src.charCodeAt(start + 1) !== 62 /* > */ ||
+      state.src.charCodeAt(start + 2) !== 104 /* h */ ||
+      state.src.charCodeAt(start + 3) !== 105 /* i */ ||
+      state.src.charCodeAt(start + 4) !== 100 /* d */ ||
+      state.src.charCodeAt(start + 5) !== 101 /* e */
+      ) {
+    return false;
+  }
+  state.pos = start + 6;
+
+  // check for end in same line
+  while (state.pos + 2 <= max) {
+    if (state.src.charCodeAt(state.pos) === 60 /* < */ &&
+        state.src.charCodeAt(state.pos + 1) === 61 /* = */
+    ) {
+      found = true;
+      state.pos += 2;
+      break;
+    }
+
+    state.md.inline.skipToken(state);
+  }
+
+  if (!found || start + 1 === state.pos) {
+    state.pos = start;
+    return false;
+  }
+
+  content = state.src.slice(start + 6, state.pos);
+
+  // don't allow unescaped newlines inside
+  if (content.match(/(^|[^\\])(\\\\)*[\n]/)) {
+    state.pos = start;
+    return false;
+  }
+  return true;
+}
+
+/*
+** Referenced from markdown-it's fence rule and markdown-it-container plugin
+** See here: https://github.com/markdown-it/markdown-it/blob/master/lib/rules_block/fence.js
+** Outputs everything that is conditionally stated to be hidden as ''
+*/
+
+function hideShowBlock(state, startLine, endLine, silent) {
+  let category;
+  let mem;
+  let marker;
+  let len;
+  let params;
+  let nextLine;
+  let token;
+  let markup;
+  let hasCategory = false;
+  let haveEndMarker = false;
+  let pos = state.bMarks[startLine] + state.tShift[startLine];
+  let max = state.eMarks[startLine];
+
+  if (pos + 7 > max) { return false; }
+
+  // console.log(state);
+  findName(state, startLine);
+
+  marker = state.src.charCodeAt(pos);
+
+  // Check out the first character quickly,
+  // this should filter out most of non-containers
+  if (marker !== 61 /* = */) { return false; }
+
+  if (state.src.charCodeAt(pos + 1) !== 61  /* = */||
+      state.src.charCodeAt(pos + 2) !== 62 /* > */ ||
+      state.src.charCodeAt(pos + 3) !== 104 /* h */ ||
+      state.src.charCodeAt(pos + 4) !== 105 /* i */ ||
+      state.src.charCodeAt(pos + 5) !== 100 /* d */ ||
+      state.src.charCodeAt(pos + 6) !== 101 /* e */
+      ) {
+    return false;
+  }
+
+  // Check if got category e.g. ==>hide:slides ...
+  if (state.src.charCodeAt(pos + 7) === 58  /* : */) {
+    hasCategory = true;
+    category = state.src.slice(pos+8, max).toString();
+  }
+
+  mem = pos;
+  pos = state.skipChars(pos, state.src.charCodeAt(pos + 6));
+
+  len = pos - mem;
+
+  markup = state.src.slice(mem, pos);
+  params = state.src.slice(pos, max);
+
+  // Since start is found, we can report success here in validation mode
+  if (silent) { return true; }
+
+  // search end of block
+  nextLine = startLine;
+
+  if (!hasCategory || category === fileName) {
+    for (;;) {
```

Comment:
Extra care is needed when using a infinity loop. Make sure that the code can always get out of the loop in all condition. [same for below]